### PR TITLE
Support PCBB tests for Active-Active DualToR

### PR DESCRIPTION
**Pull request by rbpittman**
### Description of PR

Summary:
Added marker to several tests to enable A-A to not skip. 

Test support added for:
- test_encap_dscp_rewrite
- test_bounced_back_traffic_in_expected_queue
- test_tunnel_decap_dscp_to_queue_mapping
- test_pfc_watermark_extra_lossless_standby
- test_pfc_watermark_extra_lossless_active

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [X] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205
- [X] 202305

#### How did you verify/test it?
Tested on Cisco-8000 DualToR Active-Active.
All tests pass or skip in the tunnel tests.
test_link_drop.py has same pass-rate before/after this code. 

Tested on Active-Standby to ensure no regressions on relevant tests. 

#### Any platform specific information?

None of this code is platform-specific. It is the basic A-A fixture and call support needed to have PCBB execute on A-A. It should apply to all platforms.

### CODE REVIEW

**Comment by bingwang-ms** [comment]
@lolyu Can you please help review?

**Review by lolyu — COMMENTED** [review]
If you want one ToR `standby`, please use https://github.com/sonic-net/sonic-mgmt/blob/8aa310083c9e37b44684bcdc12bb227daf9e90d0/tests/common/dualtor/dual_tor_utils.py#L1587

**Comment by rbpittman** [comment]
Hi @lolyu, thanks for the feedback.
PR's shorter now, ready for re-review!

**Review by alpeshspatel — APPROVED** [review]

**Comment by XuChen-MSFT** [comment]
Hi @lolyu,  can you help re-review ?

File: tests/qos/test_tunnel_qos_remap.py [hunk 1]
```diff
@@ -68,8 +71,10 @@ def _last_port_in_last_lag(lags):
     return lags[last_lag][-1]
 
 
-def test_encap_dscp_rewrite(ptfhost, upper_tor_host, lower_tor_host,                            # noqa F811
-                            toggle_all_simulator_ports_to_lower_tor, tbinfo, ptfadapter, tunnel_qos_maps):       # noqa F811
+@pytest.mark.enable_active_active
```
Comment (lolyu) [inline tests/qos/test_tunnel_qos_remap.py:74]: no need to add this mark

File: tests/qos/test_tunnel_qos_remap.py [hunk 2]
```diff
@@ -68,8 +71,10 @@ def _last_port_in_last_lag(lags):
...
+def test_encap_dscp_rewrite(ptfhost, upper_tor_host, lower_tor_host,  # noqa F811
```
Comment (lolyu) [inline tests/qos/test_tunnel_qos_remap.py:75]: maybe it is better to add a new fixture to setup the active-active ports into active-standy mode.

File: tests/qos/test_tunnel_qos_remap.py [hunk 3]
```diff
@@ -131,9 +137,11 @@ def test_encap_dscp_rewrite(ptfhost, upper_tor_host, lower_tor_host,
     logger.info("Verified {} DSCP inner->outer combinations".format(len(DSCP_COMBINATIONS)))
 
 
+@pytest.mark.enable_active_active
```
Comment (lolyu) [inline tests/qos/test_tunnel_qos_remap.py:140]: same as above

File: tests/qos/test_tunnel_qos_remap.py [hunk 4]
```diff
@@ -187,9 +196,11 @@ def test_bounced_back_traffic_in_expected_queue(ptfhost, upper_tor_host, lower_t
                       "The queue counter for DSCP {} Queue {} is not as expected".format(dscp, queue))
 
 
+@pytest.mark.enable_active_active
```
Comment (lolyu) [inline tests/qos/test_tunnel_qos_remap.py:199]: same as above

File: tests/qos/test_tunnel_qos_remap.py [hunk 5]
```diff
@@ -502,8 +514,12 @@ def test_pfc_pause_extra_lossless_active(ptfhost, fanouthosts, rand_selected_dut
             time.sleep(5)
 
 
+@pytest.mark.enable_active_active
```
Comment (lolyu) [inline tests/qos/test_tunnel_qos_remap.py:517]: same as above

File: tests/qos/test_tunnel_qos_remap.py [hunk 6]
```diff
@@ -600,8 +617,12 @@ def test_pfc_watermark_extra_lossless_standby(ptfhost, fanouthosts, rand_selecte
                 queue_wmk, required_wmk_bytes)
 
 
+@pytest.mark.enable_active_active
```
Comment (lolyu) [inline tests/qos/test_tunnel_qos_remap.py:620]: same as above

File: tests/qos/test_tunnel_qos_remap.py (hunk 1 above)
Comment (rbpittman) [reply tests/qos/test_tunnel_qos_remap.py]: Moved to common location at top of file.

File: tests/qos/test_tunnel_qos_remap.py (hunk 2 above)
Comment (rbpittman) [reply tests/qos/test_tunnel_qos_remap.py:76]: Added common fixtures for the different use cases present in this file.

**Comment by rbpittman** [comment]
Hi @lolyu, @XuChen-MSFT ,
Made a couple changes per your comments:
- Moved marker to be common for the entire file to reduce redundancy.
- Created common fixtures to avoid excess function invocations in the code.

**Review by lolyu — APPROVED** [review]

**Comment by rbpittman** [comment]
@XuChen-MSFT Can you help with write-access approval/merge?

**Comment by vivekverma-arista** [comment]
This has broken the following testcases for active-standby dualtor -

```
1. qos/test_tunnel_qos_remap.py::test_encap_dscp_rewrite[active-standby]
2. qos/test_tunnel_qos_remap.py::test_bounced_back_traffic_in_expected_queue[active-standby]
3. qos/test_tunnel_qos_remap.py::test_tunnel_decap_dscp_to_queue_mapping
```

```
    @pytest.fixture
    def toggle_all_aa_ports_to_lower_tor(config_active_active_dualtor_active_standby,
                                         lower_tor_host, upper_tor_host, active_active_ports):  # noqa F811
&gt;       config_active_active_dualtor_active_standby(lower_tor_host, upper_tor_host, active_active_ports)
E       TypeError: 'NoneType' object is not callable
```

```
    @pytest.fixture
    def toggle_all_aa_ports_to_lower_tor(config_active_active_dualtor_active_standby,
                                         lower_tor_host, upper_tor_host, active_active_ports):  # noqa F811
&gt;       config_active_active_dualtor_active_standby(lower_tor_host, upper_tor_host, active_active_ports)
E       TypeError: 'NoneType' object is not callable
```

```
    @pytest.fixture
    def toggle_all_aa_ports_to_rand_selected_tor(config_active_active_dualtor_active_standby,
                                                 rand_selected_dut, rand_unselected_dut, active_active_ports):  # noqa F811
&gt;       config_active_active_dualtor_active_standby(rand_selected_dut, rand_unselected_dut, active_active_ports)
E       TypeError: 'NoneType' object is not callable
```
This is because the check for active-active ports is missing similar to this: https://github.com/sonic-net/sonic-mgmt/blob/master/tests/common/dualtor/dual_tor_utils.py#L1807

**Comment by vivekverma-arista** [comment]
> This has broken the following testcases for active-standby dualtor in 202305, 202311 and master -
> 
> ```
> 1. qos/test_tunnel_qos_remap.py::test_encap_dscp_rewrite[active-standby]
> 2. qos/test_tunnel_qos_remap.py::test_bounced_back_traffic_in_expected_queue[active-standby]
> 3. qos/test_tunnel_qos_remap.py::test_tunnel_decap_dscp_to_queue_mapping
> ```
> 
> ```
>     @pytest.fixture
>     def toggle_all_aa_ports_to_lower_tor(config_active_active_dualtor_active_standby,
>                                          lower_tor_host, upper_tor_host, active_active_ports):  # noqa F811
> &gt;       config_active_active_dualtor_active_standby(lower_tor_host, upper_tor_host, active_active_ports)
> E       TypeError: 'NoneType' object is not callable
> ```
> 
> ```
>     @pytest.fixture
>     def toggle_all_aa_ports_to_rand_selected_tor(config_active_active_dualtor_active_standby,
>                                                  rand_selected_dut, rand_unselected_dut, active_active_ports):  # noqa F811
> &gt;       config_active_active_dualtor_active_standby(rand_selected_dut, rand_unselected_dut, active_active_ports)
> E       TypeError: 'NoneType' object is not callable
> ```
> 
> This is because the check for active-active ports is missing similar to this: https://github.com/sonic-net/sonic-mgmt/blob/master/tests/common/dualtor/dual_tor_utils.py#L1807

https://github.com/sonic-net/sonic-mgmt/pull/13001 raised to fix this.

**Pull request merged**